fw/preamp/src/i2c.c

```diff
@@ -75,6 +75,51 @@ void initI2C2() {
   I2C_Cmd(I2C2, ENABLE);
 }
 
+void deinitI2C2() {
+  /* Disable I2C2 and release GPIO control */
+
+  // Ensure I2C clock is enabled
+  RCC_APB1PeriphClockCmd(RCC_APB1Periph_I2C2, ENABLE);
```

Comment:
why does the clock need to be enabled before i2c is disabled?

Reply:
This is the peripheral clock, can't write to any of the peripheral's registers if it's disabled. `I2C_Cmd(I2C2, DISABLE);` writes to an I2C register to disable it. Technically right now `deinitI2C2()` isn't even needed since it's being called before I2C is initialized (hence why the peripheral's clock isn't enabled), but there was a thought to further robustify things in the future and if a `I2C_ISR_ARLO` is ever thrown call the `quiesceI2C()` function. This would de-init the bus, send extra clocks to try and resolve the situation, then re-init the bus.

Reply:
Some part of this needs to be in a comment in there, but you already know that :)